core/variable_operations.cpp

```diff
@@ -294,4 +294,52 @@ Variable masks_merge_if_contained(const MasksConstProxy &masks,
   return mask_union;
 }
 
+VariableProxy replace_nan(const VariableConstProxy &var, double replacement,
+                          const VariableProxy &out) {
+
+  if (var.dtype() != out.dtype())
+    throw except::TypeError("Input and output variable types do not match");
+  if (var.dtype() != dtype<decltype(replacement)>)
+    throw except::TypeError("Replacement type doesn't match type of input");
```

Comment:
This implies the the algorithm can only work for `double`. Maybe we could pass the replacement as a `Variable`?

Reply:
Yes, passing single value variable makes sense. I hadn't considered that, I was instead considering making this a templated argument.